pkg/validations/proxy_validation.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache v2.0 license.
+
+package validations
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/microsoft/moc/pkg/errors"
+)
+
+func ValidateProxyURL(proxyURL string, certContent string) error {
```

Comment:
When/Where are we going to call this function?

Reply:
in moc sdk and cloud agent for url validation